api/app_analytics/models.py

```diff
@@ -0,0 +1,93 @@
+from datetime import date
+
+from django.db import models
+from django.utils import timezone
+
+
+class Resource(models.IntegerChoices):
+    FLAGS = 1
+    IDENTITIES = 2
+    TRAITS = 3
+    ENVIRONMENT_DOCUMENT = 4
+
+
+class APIUsageRaw(models.Model):
+    environment_id = models.PositiveIntegerField()
+    created_at = models.DateTimeField(auto_now_add=True)
+    host = models.CharField(max_length=255)
+    resource = models.IntegerField(choices=Resource.choices)
+
+    class Meta:
+        index_together = (("environment_id", "created_at"),)
+
+
+class APIUsageAggBucketWindow(models.IntegerChoices):
+    HOUR = 1
+    DAY = 2
+
+
+class APIUsageBucket(models.Model):
+    environment_id = models.PositiveIntegerField()
+    resource = models.IntegerField(choices=Resource.choices)
+    total_count = models.PositiveIntegerField()
+    created_at = models.DateTimeField(auto_now_add=True)
+    bucket_size = models.PositiveIntegerField(help_text="Bucket size in minutes")
+
+
+# TODO: move to tasks
+def populate_bucket(bucket_size: int):
+    now = timezone.now()
+    process_from = now - timezone.timedelta(minutes=bucket_size + 1)
+    data = (
+        APIUsageRaw.objects.filter(created_at__gt=process_from)
```

Comment:
Here we'd need to aggregate by date somehow as we want to process more than one time window each time the task runs - this is only creating the data for a single time window. 

Could we use the `Extract` or `Trunc` functions of the Django ORM perhaps? 

https://docs.djangoproject.com/en/3.2/ref/models/database-functions/#extract
https://docs.djangoproject.com/en/3.2/ref/models/database-functions/#trunc